packages/admin-stories/src/docs/components/Selection/Selection.stories.mdx

```diff
@@ -0,0 +1,110 @@
+-   `mode` specifies the goal of the selection. Either `add` (add a new item) or `edit` (edit an existing item).
+
+The second object is the [Selection API](#selection-api).
+
+<Canvas>
+    <Story id="stories-components-selection-useselection-hook--useselection-hook" />
+</Canvas>
+
+### Selection Component
+
+The `Selection` component passes its values to its children via child function.
+
+It provides the same values as `useSelection()`, but the variable names are slightly different:
+
+-   `selectedId` instead of `id`
+-   `selectionMode` instead of `mode`
+-   `selectionApi` contains the [Selection API](#selection-api)
+
+<Canvas>
+    <Story id="stories-components-selection-selection-component--selection-component" />
+</Canvas>
+
+## Selection with URL State
+
+In practice, a common requirement is that each item should have a unique URL.
+For that purpose, the id of the currently selected item has to be written to the URL path.
+Comet Admin has this functionality built-in.
+
+To implement a selection with URL state, you can use the `useSelectionRoute()` hook or the `SelectionRoute` component.
+
+### useSelectionRoute() Hook (Recommended)
+
+The `useSelectionRoute()` hook is similar to `useSelection()`.
+The main difference is that it returns three objects instead of two.
+
+The first - and additional - object is now a `SelectionRoute`.
+This `SelectionRoute` must wrap the selection-specific part of your application for the routing to function correctly.
+
+<Canvas>
+    <Story id="stories-components-selection-useselectionroute-hook--useselectionroute-hook" />
+</Canvas>
+
+### SelectionRoute Component
+
+The `SelectionRoute` is the URL-stateful equivalent of the `Selection` component.
+You can use it exactly like `Selection` since it provides the same values.
+
+<Canvas>
+    <Story id="stories-components-selection-selectionroute-component--selectionroute-component" />
+</Canvas>
+
+## Selection API
+
+The Selection API provides three functions to select and deselect items:
+
+-   `handleSelectId()` for selecting an existing item
+-   `handleAdd()` for "selecting" a new item. It accepts an optional `id` parameter if you generate the new id on the client-side.
+-   `handleDeselect()` for deselection
+
+## Selected
+
+The `Selected` component gives you direct access to the item that matches a selected id.
+
+In the simplest configuration, you have to provide the props `selectedId`, `selectionMode` and `rows`.
+`rows` must be an array containing all items.
+The `Selected` component uses this array to find the selected item based on its id.
+It then provides the item to its children via child function.
+
+<Canvas>
+    <Story id="stories-components-selection-selected--selected" />
+</Canvas>
+
+### Selected with Lazy Loading
+
+If you don't want to preload all item data, you can lazy load each item once it's been selected.
+To do that, you have to provide the two props: `query` and `dataAccessor`.
```

Comment:
For the use case you describe, Selected should not be used, just use apollo.

The use case where lazy loading of Selected should be used is the following:

- you load all item data in the table
- and you have paging
- then it can happen (eg. on page reload) that the table loads the first page but the selected id is on the second. Then (any only then) Selected falls back to a lazy query.

(oh, and generally I think in pratice it is usually no issue to lazy load and avoid usage of Selected, maybe you should mention that)

Reply:
> oh, and generally I think in pratice it is usually no issue to lazy load and avoid usage of Selected, maybe you should mention that

Does this mean we recommend not using `Selected` **at all** or we do not recommend using `Selected` **in situations with lazy loading**?

Reply:
@nsams

Reply:
>  It sounds like a very specific edge case.

When you have paging (and use Selected) you have this issue and need to solve it somehow.

> Does this mean we recommend not using Selected at all or we do not recommend using Selected in situations with lazy loading?

I would not recommend using Selected at all. It's purpose (avoid lazy loading) does solve a problem that is none. And it over-complicates things.

For me it is Ok to not document Selected at all.

Reply:
@nsams Okay, I removed the docs for `Selected` completely

If we don't recommend using `Selected` anymore, we could mark it as deprecated and remove it in v3 (or later)